m2cgen/interpreters/mixins.py

```diff
@@ -195,3 +195,46 @@ def _get_subroutine_name(self):
 
     def create_code_generator(self):
         raise NotImplementedError
+
+
+class InfixPowExprMixin(BaseToCodeInterpreter):
+    """
+    This mixin is used for languages that provide the exponentiation operation in a form
+    of an infix operator. Such languages inlcude Haskell, F#, R and others.
+
+    The operator used by default is "**", but it can be overriden by setting an appropriate
+    value in `pow_operator`.
+    """
+
+    pow_operator = "**"
+
+    infix_expressions = [*BaseToCodeInterpreter.infix_expressions, ast.PowExpr]
```

Comment:
See comment above about eliminating `list -> tuple` cast for some speedup.
```suggestion
    infix_expressions = (*BaseToCodeInterpreter.infix_expressions, ast.PowExpr)
```

Reply:
I actually didn't know this syntax was supported! Otherwise I wouldn't have introduced the list->tuple conversion. TIL!